acvm-repo/acvm/src/compiler/transformers/csat.rs

```diff
@@ -461,7 +461,25 @@ fn fits_in_one_identity<F: AcirField>(expr: &Expression<F>, width: usize) -> boo
         }
     }
 
-    found_x & found_y
+    // If the multiplication is a squaring then we must assign the two witnesses to separate wires and so we
+    // can never get a zero contribution to the width.
+    let multiplication_is_squaring = mul_term.1 == mul_term.2;
+
+    let mul_term_width_contribution = if !multiplication_is_squaring && (found_x & found_x) {
```

Comment:
I think it should be `found_x & found_y`

Reply:
Thanks! Yeah, clippy pointed this out as well so all's fixed now.